### ISSUE TITLE
in webview preoload js ,can't receive msg from main immediately

### ISSUE THREAD

**Issue by spcBackToLife** [electron/electron#27137]
electron: v11.1.1 
os: mac: 10.15.7

## demo 1:
in preload js, just send msg to main,and register listener on 'ipc:connect' channel, can only receive msg from 10ms later.

- main.js
```javascript
const { app, BrowserWindow, ipcMain } = require('electron');

app.on('ready', () => {
  const win = new BrowserWindow({
    width: 800,
    height: 800,
    webPreferences: {
      enableRemoteModule: true,
      nodeIntegration: true,
      webviewTag: true,
    }
  })

  win.loadURL(`file://${__dirname}/index.html`);
  ipcMain.on('ipc:hello', (e, msg) => {
    console.log('main:', msg);
    e.sender.send('ipc:connect', '1');
     setTimeout(() => {
       e.sender.send('ipc:connect', '2');
     }, 10)
  });
})

```

- html
```
<!DOCTYPE html>
<html>
  <body>
    123
    <div>
      <webview
        src="https://www.baidu.com"
        id="webview1"
        style="width: 800px;height: 800px;"
        preload="file:///Users/pikun/Documents/test/test-electron-ipc-use-electron/preload.js">
      </webview>
    </div>
    <script>
      const webview = document.querySelector('webview');
      if (webview) {
        webview.addEventListener('dom-ready', () => {
        console.log('ready');
        webview.openDevTools();
      })
      }
    </script>
  </body>
</html>

```

- preloadjs

```javascript
const { ipcRenderer } = require('electron');
console.log('this is prepload');
ipcRenderer.send('ipc:hello', 'from webview');
ipcRenderer.on('ipc:connect', (e, msg) => {
  console.log('this is msg from main:', msg);
})
```
so, only print msg of "2" of this:
```javascript
  setTimeout(() => {
       e.sender.send('ipc:connect', '2');
     }, 10)
```

but this cannot received:
```
e.sender.send('ipc:connect', '1');
```

## Demo 2
just the same as before, but main:
```javascipt
 ipcMain.on('ipc:hello', (e, msg) => {
    console.log('main:', msg);
    e.sender.send('ipc:connect', '1');
  });
```

and preload:
```
const { ipcRenderer } = require('electron');

setTimeout(() => {
  ipcRenderer.send('ipc:hello', 'from webview');
  ipcRenderer.on('ipc:connect', (e, msg) => {
  console.log('this is msg from main:', msg);
})
}, 10)
```

this can also receive msg, so why have to delay 10ms.

expect:
when preloadjs send msg to main process, can receive msg immediately, do not need to delay 10ms !!!!

**Comment by hxkuc** [comment]
same question

**Comment by thorsent** [comment]
+1 Still exists on Electron 13.1.9

**Comment by spcBackToLife** [comment]
> +1 Still exists on Electron 13.1.9

it seems fixed in "electron": "13.1.9", I have tried.

before 13.1.9 . browserWindow also has the same problem

**Comment by spcBackToLife** [comment]
still have bug in Electron 13.1.9. after load a url, continue to load another url. the problem still exists!
